### PR TITLE
fix putting larger mixed qualifier structs in std.variant

### PR DESCRIPTION
**Pull request by John-Colvin**

### CODE REVIEW

File: std/variant.d [hunk 1]
```diff
@@ -672,33 +673,39 @@ public:
...
                 // not really shared, as it's copied out as well.
-                memcpy(&store, cast(const(void*)) &rhs, rhs.sizeof);
+                memcpy(dest, src, len);
                 static if (hasElaborateCopyConstructor!T)
                 {
                     // Safer than using typeid's postblit function because it
                     // type-checks the postblit function against the qualifiers
                     // of the type.
-                    (cast(T*)&store).__xpostblit();
+                    (cast(T*)dest).__xpostblit();
```
Comment (MoonlightSentinel) [inline std/variant.d]:
```suggestion
                    (cast(T*) dest).__xpostblit();
```

File: std/variant.d [hunk 2]
```diff
@@ -1356,6 +1363,19 @@ public:
     assert(v == S.init);
 }
 
+@system unittest
```
Comment (MoonlightSentinel) [inline std/variant.d]: Please add a comment with the issue number

**Comment by RazvanN7** [comment]
ping @John-Colvin

**Comment by RazvanN7** [comment]
The unittest added in this PR currently compiles with the latest master of phobos. Given that no issue is linked and the lack of activity, I will close this PR. @John-Colvin feel free to reopen at any time.

**Comment by John-Colvin** [comment]
@rmanthorpe i can't remember, maybe this was actually one of your changes that I upstreamed?

**Comment by rmanthorpe** [comment]
No this was all yours. I swapped out typeinfo for a mangle because windows, DLLs.

**Pull request closed without merging**